operational_analysis/methods/quality_check_automation.py

```diff
@@ -1,67 +1,181 @@
+"""Creates the QualityControlDiagnosticSuite and specific data-based subclasses."""
+
+from abc import abstractmethod
+from typing import List, Tuple, Union
+from datetime import time, datetime
+
+import pytz
 import h5pyd
 import numpy as np
 import pandas as pd
 import dateutil
 import matplotlib.pyplot as plt
 from pyproj import Proj
+from dateutil import tz
 
 from operational_analysis import logging, logged_method_call
-from operational_analysis.toolkits import filters, timeseries, power_curve
+from operational_analysis.toolkits import timeseries
 
 
+Number = Union[int, float]
 logger = logging.getLogger(__name__)
 
 
-class WindToolKitQualityControlDiagnosticSuite(object):
+def _read_data(data: Union[pd.DataFrame, str]) -> pd.DataFrame:
+    """Takes the `DataFrame` or file path and returns a `DataFrame`
+
+    Args:
+     data(:obj: `Union[pd.DataFrame, str]`): The actual data or a path to the csv data.
+
+    Returns
+     (:obj: `pd.DataFrame`): The data fram object.
+    """
+    if isinstance(data, pd.DataFrame):
+        return data
+    return pd.read_csv(data)
+
+
+def _remove_tz(df: pd.DataFrame, t_local_column: str) -> Tuple[np.ndarray, np.ndarray]:
+    """Identify the non-timestamp elements in the DataFrame timestamp column and return
+    a truth array for filtering the values and the timezone-naive timestamps.
+
+    Args:
+        df (:obj:`pandas.DataFrame`): The DataFrame of interest.
+        t_local_column (:obj:`str`): The name of the timestamp column.
+
+    Returns:
+        :obj:`numpy.ndarray`: Truth array that can be used to filter the timestamps and subsequent values.
+        :obj:`numpy.ndarray`: Array of timezone-naive python `datetime` objects.
+    """
+    arr = np.array(
+        [
+            [True, pd.to_datetime(el).tz_localize(None).to_pydatetime()]
+            if not isinstance(el, float)
```

Comment:
In theory yes, but in practice, this function is more geared toward internal data processing where at least in the issues i've faced having a float at this stage is where we run into trouble. I'll at least update the docstring so it can be more properly used in expansion efforts here.